src/tools/run-make-support/src/symbols.rs

```diff
@@ -0,0 +1,37 @@
+use object::{self, Object, ObjectSymbol, SymbolIterator};
+use std::fs;
+use std::path::Path;
+
+/// iterate through the symbols in an object file.
+///
+/// uses a callback because SymbolIterator does not own its data
+pub fn with_symbol_iter<P, F, R>(path: P, func: F) -> R
+where
+    P: AsRef<Path>,
+    //I: Iterator + 'a,
```

Comment:
Cleanup nit

```suggestion
/// Iterate through the symbols in an object file.
///
/// Uses a callback because SymbolIterator does not own its data.
pub fn with_symbol_iter<P, F, R>(path: P, func: F) -> R
where
    P: AsRef<Path>,
```